DH_Engine/Classes/DHCommandMenu_FireSupport.uc

```diff
@@ -0,0 +1,320 @@
+//==============================================================================
+// Darkest Hour: Europe '44-'45
+// Darklight Games (c) 2008-2019
+//==============================================================================
+
+class DHCommandMenu_FireSupport extends DHCommandMenu;
+
+enum EArtilleryStatus
+{
+    DisabledGlobally,
+    DisabledNotEnoughMembers,
+    Enabled
+};
+
+function OnSelect(int Index, vector Location)
+{
+    local DHPlayer PC;
+    local DHPlayerReplicationInfo PRI;
+    local DHGameReplicationInfo GRI;
+    local vector MapLocation;
+
+    PC = GetPlayerController();
+
+    PRI = DHPlayerReplicationInfo(PC.PlayerReplicationInfo);
+    GRI = DHGameReplicationInfo(PC.GameReplicationInfo);
+
+    GRI.GetMapCoords(Location, MapLocation.X, MapLocation.Y);
+
+    if (PC == none || Index < 0 || Index >= Options.Length)
+    {
+        return;
+    }
+
+    PC.ServerIsArtilleryTargetValid(Location);
+    if (PC.IsArtillerySpotter() && PC.bIsArtilleryTargetValid)
+    {
+        switch (Index)
+        {
+            case 0: // Artillery barrage
+                AddNewArtilleryRequest(PC, MapLocation, Location, class'DH_Engine.DHMapMarker_FireSupport_OffMap');
+                break;
+            case 1: // Fire request (Smoke)
+                AddNewArtilleryRequest(PC, MapLocation, Location, class'DH_Engine.DHMapMarker_FireSupport_Smoke');
+                break;
+            case 2: // Fire request (HE)
+                AddNewArtilleryRequest(PC, MapLocation, Location, class'DH_Engine.DHMapMarker_FireSupport_HE');
+                break;
+        }
+    }
+    else
+    {
+        PC.ReceiveLocalizedMessage(class'ROArtilleryMsg', 5); // "Not a Valid Artillery Target!"
+    }
+
+    Interaction.Hide();
+}
+
+function OnPush()
+{
+    local DHPlayer PC;
+
+    PC = GetPlayerController();
+
+    if (PC == none)
+    {
+        return;
+    }
+
+    if (PC.SpottingMarker != none)
+    {
+        PC.SpottingMarker.Destroy();
+    }
+
+    PC.SpottingMarker = PC.Spawn(class'DHSpottingMarker', PC);
+}
+
+function OnPop()
+{
+    local DHPlayer PC;
+
+    PC = GetPlayerController();
+
+    if (PC == none || PC.SpottingMarker == none)
+    {
+        return;
+    }
+
+    PC.SpottingMarker.Destroy();
+}
+
+function Tick()
+{
+    local DHPlayer                PC;
+    local vector                  HitLocation, HitNormal;
+    local Color                   C;
+    local bool                    bArtillerySupportEnabled;
+    local DHGameReplicationInfo   GRI;
+
+    bArtillerySupportEnabled = false;
+
+    C.R = 0;
```

Comment:
If I recall correctly, these values should already by `0` by default, so this shouldn't be necessary.